tests/ui/lint/non-local-defs/exhaustive.rs

```diff
@@ -57,18 +55,13 @@ fn main() {
 
     struct InsideMain;
 
+    impl Trait for &InsideMain {}
     impl Trait for *mut InsideMain {}
-    //~^ WARN non-local `impl` definition
     impl Trait for *mut [InsideMain] {}
-    //~^ WARN non-local `impl` definition
     impl Trait for [InsideMain; 8] {}
-    //~^ WARN non-local `impl` definition
     impl Trait for (InsideMain,) {}
-    //~^ WARN non-local `impl` definition
     impl Trait for fn(InsideMain) -> () {}
-    //~^ WARN non-local `impl` definition
     impl Trait for fn() -> InsideMain {}
-    //~^ WARN non-local `impl` definition
```

Comment:
@traviscross can you confirm that those changes here and in other tests are indeed what T-lang wants?

Reply:
The changes to the tests look consistent with the decision to me.